### PR TITLE
Kan 17 fully swap static data into data from db

### PR DESCRIPTION
**Pull request by Ech0ke**
Data is now from db. **Currently there is no refresh when creating/editing note, so make sure you manually add folder for your userID according to this example in firestore**. Make sure to create new user, since we are now on v3 and not v2 anymore.
image

By going into folder and navigating back, you will trigger refetch and you should see new/updated notes in **browse** tab.

You can test the functionality

- Create note
- Edit note
- Nested folders

I will now start implementing ability to create folders from the app and will probably add live data refresh or at least pull up gesture to refresh (refetch).

### CODE REVIEW

File: src/app/(tabs)/index.tsx [hunk 1]
```diff
@@ -2,14 +2,33 @@ import { auth } from "@/firebaseConfig"
...
+    return (
+      <View>
+        <Text>Hi</Text>
+      </View>
+    )
+
   return (
     <SafeAreaView className="flex-1">
-      <Text>Welcome {user?.getIdToken}</Text>
+      <Text>Welcome {username}</Text>
```
Comment (matuliux) [inline src/app/(tabs)/index.tsx:31]: Here I get `Welcome {my jwt token}`
I dont think this is a big issue but just wanna check if you intended that

**Review by matuliux — COMMENTED** [review]
I have added a new folder into db
image

and assigned a note to it

image

But after reloading the app I dont see neither the note nor the folder

image

Can you spot why?

**Review by matuliux — COMMENTED** [review]
Also I think this part should not be here. I have had it removed with my previous PR just not sure if changes got lost or if they were reverted

image

Edit: I added this part into my upcoming PR so you can leave it as is


image

**Comment by Ech0ke** [comment]
@matuliux 
I've fixed folder fetching, refactored auth according to expo documentation and now upon reloading app, you don't need to relogin, since auth state is saved.

try to run it now but there may be cache left, so after pulling the changes, run this command: `npx expo start --tunnel --clear`

**Review by matuliux — APPROVED** [review]
Got folders working (issue from before seems to be fixed)
Everything works great!
Also I haven't noticed any flickering yet, not even when typing big notes. If you get this bug please send a screen recording so I could try to recreate it

**Pull request merged**